library/bignum.c

```diff
@@ -158,9 +158,10 @@ int mbedtls_mpi_shrink( mbedtls_mpi *X, size_t nblimbs )
     if( nblimbs > MBEDTLS_MPI_MAX_LIMBS )
         return( MBEDTLS_ERR_MPI_ALLOC_FAILED );
 
-    /* Actually resize up in this case */
+    /* Actually resize up if there are currently fewer than nblimbs limbs. */
     if( X->n <= nblimbs )
         return( mbedtls_mpi_grow( X, nblimbs ) );
+    /* Now X->n > nblimbs >= 0. */
```

Comment:
Minor: While we're improving comments, I'd like to suggest replacing "Now" with "If we reach this point, we know that". Initially I missed that there there was a `return` on the line above, so I started commenting that we only know `X->n >= nblimbs` here before realizing my mistake. I think wording like the one I suggested would have helped draw my attention to the `return` above.